### ISSUE TITLE
gaia with Qt5

### ISSUE THREAD

**Issue by anthonyhuecouret** [MTG/gaia#28]
Hello,
I'm building an application in which I would like to use essentia ( so gaia required ) but this app uses Qt5.
I'm encountering a problem because gaia depends on Qt4, I read this post  https://github.com/MTG/gaia/issues/7 saying that you're or will be working on an Qt free version. Waiting for that I tried to build gaia with Qt5 but got tons of errors with Qt globals not being defined or something. I got messages "Expected ';' at end of declaration list" for every Qt function declaration ending with a Q_DECL_* macro. 
Does anybody have guidelines or successfully built gaia with Qt5 ?
Thanks

**Comment by dbogdanov** [comment]
Removing the dependency on Qt is in our wish-list, but it is not of the primary importance, so I doubt it will be done in the closest future. Can you work with both qt4 and qt5 simultaneously?

**Comment by anthonyhuecouret** [comment]
I could be wrong but I think going with both in the same project is not possible. But I have the answer to the issue. So the macro stuff was my mistake because '/usr/local/include' is added to the INCLUDEPATH variable in Gaia2lib.pro and I actually have qt4 headers in this directory, that's why there were some conflicts. 

I'll give the modifications I did to get gaia built with qt5 :

replace all instances of toAscii() to toLatin1()
[ yamlcpp.h : 106, utils.cpp : 697 filter.cpp : 213 keydistance.cpp : 30, 43 svmtrain.cpp : 186, 198 ]

replace all instances of qMemCopy to memcpy 
[ gvarlengtharray.h : 202, 227 ]

modify the qtMessageHandler function in gaia.cpp with : 

static void qtMessageHandler
(QtMsgType type, const QMessageLogContext & log_context, const QString & msg) {
    fprintf(stderr, "%s\n", msg.toLatin1().data() );
    if (type == QtFatalMsg) { abort(); }
}

then replace qInstallMsgHandler(qtMessageHandler); to qInstallMessageHandler(qtMessageHandler);

add #include < QDataStream > in types.h

finally in Gaia2lib.pro :

add the concurrent module : QT += concurrent

I simply commented the inclusions of '/usr/local/include' and added the include path for yaml header file.

:-)
